File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int n=nums.length;
        int L=0;
        int R=n-1;

        while(L<=R){
            int M=(L+R)/2;

            if(target == nums[M]){
                return M;
            }
            else if(target < nums[M]){
                R=M-1;
            }
            else{
                L=M+1;
            }
        }
        return -1;
    }
}

File: 792-binary-search/binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        n=len(nums)
        l=0
        r=n-1
        
        while(l<=r):
            m=(l+r)//2

            if target == nums[m]:
                return m
            elif target < nums[m]:
                r=m-1
            else:
                l=m+1
        return -1